source/testing/index.md

```diff
@@ -1,79 +1,146 @@
+such as behavior between UI controls.
+Typically they require an UI to be rendered to be able to execute user interactions in the testing scenario.
+They are valuable in ensuring data and actions are properly passed between different parts of the system,
+and provide confidence that parts of the system will work within the application under multiple scenarios.
 
-* A fullname attribute is computed which combines the `firstName` and `lastName` attributes.
-* The serializer properly converts the blog request payload into a blog post model object.
-* Blog dates are properly formatted.
-
-### Integration Tests
-
-Integration tests serve as a middle ground between acceptance tests, which only interact
-with the full system through user endpoints, and unit tests, which interact with specific
-code algorithms on a micro level. Integration tests verify interactions between various
-parts of the application, such as behavior between UI controls.  They are valuable
-in ensuring data and actions are properly passed between different parts of the system, and
-provide confidence that parts of the system will work within the application under multiple
-scenarios.
-
-It is recommended that components and helpers be tested with integration tests because the component
-interacts with the system in the same way that it will within the context of the application,
+Rendering tests are especially useful for verifying the correct behavior of components and helpers.
+The component or helper interacts with the system in the same way that it will within the context of the application,
 including being rendered from a template and receiving Ember's lifecycle hooks.
 
-Examples of integration tests are:
+Examples of these kind of tests are:
 
 * An author's full name and date are properly displayed in a blog post.
 * A user is prevented from typing more than 50 characters into post's title field.
 * Submitting a post without a title displays a red validation state on the field and gives the user text indicating that the title is required.
 * The blog post list scrolls to position a new post at the top of the viewport.
 
+We can create an isolated rendering test with the `setupRenderingTest` helper.
+You can read more about it in the [Testing Components] or the [Testing Helpers] section.
+
+### Container Tests
+
+Container tests are useful if you are testing a small part of your app which doesn't require user interaction to be tested
+but which requires the application's container to be setup.
+If you would like to test a functionality regarding an Ember class instance,
+e.g. a controller, service or route, you can create this kind of test to verify its behavior.
+
+Some specific examples of these type of tests are:
+
+* A `fullName` attribute on a controller is computed by combining its `firstName` and `lastName` attributes.
+* The serializer properly converts the blog request payload into a blog post model object.
```

Comment:
Is a serializer a good example to use here? I've had headaches testing this in the old system, so if it is, hallelujah! 😀 Otherwise, we might not want to specifically mention it if folks are still going to have issues

Reply:
Also, probably should be *A* serializer